crates/iceberg/src/spec/view_version.rs

```diff
@@ -0,0 +1,347 @@
+    timestamp_ms: i64,
+    /// A string to string map of summary metadata about the version
+    summary: HashMap<String, String>,
+    /// A list of representations for the view definition.
+    representations: ViewRepresentations,
+    /// Catalog name to use when a reference in the SELECT does not contain a catalog
+    #[builder(default = None)]
+    default_catalog: Option<String>,
+    /// Namespace to use when a reference in the SELECT is a single identifier
+    default_namespace: NamespaceIdent,
+}
+
+impl ViewVersion {
+    /// Get the version id of this view version.
+    #[inline]
+    pub fn version_id(&self) -> i64 {
+        self.version_id
+    }
+
+    /// Get the schema id of this view version.
+    #[inline]
+    pub fn schema_id(&self) -> SchemaId {
+        self.schema_id
+    }
+
+    /// Get the timestamp of when the view version was created
+    #[inline]
+    pub fn timestamp(&self) -> MappedLocalTime<DateTime<Utc>> {
+        Utc.timestamp_millis_opt(self.timestamp_ms)
+    }
+
+    /// Get the timestamp of when the view version was created in milliseconds since epoch
+    #[inline]
+    pub fn timestamp_ms(&self) -> i64 {
+        self.timestamp_ms
+    }
+
+    /// Get summary of the view version
+    #[inline]
+    pub fn summary(&self) -> &HashMap<String, String> {
+        &self.summary
+    }
+
+    /// Get this views representations
+    #[inline]
+    pub fn representations(&self) -> &ViewRepresentations {
+        &self.representations
+    }
+
+    /// Get the default catalog for this view version
+    #[inline]
+    pub fn default_catalog(&self) -> Option<&String> {
+        self.default_catalog.as_ref()
+    }
+
+    /// Get the default namespace to use when a reference in the SELECT is a single identifier
+    #[inline]
+    pub fn default_namespace(&self) -> &NamespaceIdent {
+        &self.default_namespace
+    }
+
+    /// Get the schema of this snapshot.
+    pub fn schema(&self, view_metadata: &ViewMetadata) -> Result<SchemaRef> {
+        let r = view_metadata
+            .schema_by_id(self.schema_id())
+            .ok_or_else(|| {
+                Error::new(
+                    ErrorKind::DataInvalid,
+                    format!("Schema with id {} not found", self.schema_id()),
+                )
+            })
+            .cloned();
+        r
+    }
+
+    pub(crate) fn log(&self) -> ViewVersionLog {
+        ViewVersionLog::new(self.version_id, self.timestamp_ms)
+    }
+}
+
+/// A list of view representations.
+pub type ViewRepresentations = Vec<ViewRepresentation>;
+
+/// A builder for [`ViewRepresentations`].
+pub struct ViewRepresentationsBuilder(ViewRepresentations);
+
+impl ViewRepresentationsBuilder {
+    /// Create a new builder.
+    pub fn new() -> Self {
+        Self(Vec::new())
+    }
+
+    /// Add a representation to the list.
+    ///
+    /// SQL representations dialects must be unique. If a representation with the same
```

Comment:
does this make sure to ignore the case of the dialect when comparing two dialects? If no, then it should

Reply:
No, it did not. I fixed it here:
https://github.com/c-thiel/iceberg-rust/commit/bf95eb7c5c458e01352afd9e5d6199d5c470477e

The behavior is now to ignore the case when adding, but storing the proper case internally (see the test in my commit).
Is this the desired behavior or should we just lowercase it?